components/shared/Form/Onboarding/OnboardingForm.tsx

```diff
@@ -200,34 +214,62 @@ const OnboardingForm: React.FC<OnboardingFormProps> = ({ step, session }) => {
   };
 
   useEffect(() => {
-    if (Object.keys(errors).length !== 0) {
+    const handleSingleFieldError = (prependingString: string, error: any) => {
+      let errorMsg = "";
+      if (error) {
+        errorMsg = `${prependingString} Error: ${error.message}`;
+        showError(errorMsg);
+      }
+    };
+
+    const handleFieldArrayError = (prependingString: string, error: any) => {
       let errorMsg = "";
-      console.log(errors);
-      if (errors.name) {
-        errorMsg = errors.name.message as string;
-      } else if (errors.portfolio) {
-        errorMsg = errors.portfolio.message as string;
+      if (!error) return;
+      if (Array.isArray(error)) {
+        for (let err of error) {
+          for (let key in err) {
+            errorMsg += `${prependingString} Error: ${err[key].message} `;
+            break;
+          }
+          if (errorMsg !== "") {
+            break;
+          }
+        }
+      } else {
+        errorMsg = `${prependingString} Error: ${error.message}`;
       }
       showError(errorMsg);
+    };
+
+    if (Object.keys(errors).length !== 0) {
+      handleSingleFieldError("Name", errors.name);
+      handleSingleFieldError("Portfolio", errors.portfolio);
+      handleFieldArrayError("Learning Goals", errors.learningGoals);
+      handleFieldArrayError("Knowledge Levels", errors.knowledgeLevels);
     }
   }, [errors]);
 
-  const onSubmit = (data: any) => {
-    console.log("we are submitting!");
-    console.log(data);
+  const onSubmit = async (data: any) => {
     updateUser(data);
+    const session = await getSession();
```

Comment:
Not sure we need this 'getSession'.

We could also await the update of the user before pushing them to the homepage:
```js
  const onSubmit = async (data: any) => {
    await updateUser(data);
    router.push("/");
  };
  ```